net: mqtt: use zsock_ functions

xhpohanka opened the pull request:
Using zephyr's internals zsock_ calls make mqtt library more compatible,
now it does not depend on NET_SOCKETS_POSIX_NAMES.

Signed-off-by: Jan Pohanka <xhpohanka@gmail.com>

xhpohanka committed:
net: mqtt: use zsock_ functions

Using zephyr's internals zsock_ calls make mqtt library more compatible,
now it does not depend on NET_SOCKETS_POSIX_NAMES.

Signed-off-by: Jan Pohanka <xhpohanka@gmail.com>

rlubos commented on subsys/net/lib/mqtt/mqtt_transport_socket_tcp.c:
```diff
@@ -23,7 +23,7 @@ int mqtt_client_tcp_connect(struct mqtt_client *client)
 
-	client->transport.tcp.sock = socket(broker->sa_family, SOCK_STREAM,
+	client->transport.tcp.sock = zsock_socket(broker->sa_family, SOCK_STREAM,
 					    IPPROTO_TCP);
```
This line is no longer aligned.

rlubos commented on subsys/net/lib/mqtt/mqtt_transport_socket_tcp.c:
```diff
@@ -49,10 +49,10 @@ int mqtt_client_tcp_connect(struct mqtt_client *client)
 
-	ret = connect(client->transport.tcp.sock, client->broker,
+	ret = zsock_connect(client->transport.tcp.sock, client->broker,
 		      peer_addr_size);
```
Same here.

rlubos commented on subsys/net/lib/mqtt/mqtt_transport_socket_tcp.c:
```diff
@@ -67,7 +67,7 @@ int mqtt_client_tcp_write(struct mqtt_client *client, const uint8_t *data,
 	while (offset < datalen) {
-		ret = send(client->transport.tcp.sock, data + offset,
+		ret = zsock_send(client->transport.tcp.sock, data + offset,
 			   datalen - offset, 0);
```
And here.

rlubos commented on subsys/net/lib/mqtt/mqtt_transport_socket_tls.c:
```diff
@@ -24,7 +24,7 @@ int mqtt_client_tls_connect(struct mqtt_client *client)
 
-	client->transport.tls.sock = socket(broker->sa_family,
+	client->transport.tls.sock = zsock_socket(broker->sa_family,
 					    SOCK_STREAM, IPPROTO_TLS_1_2);
```
Here too.

rlubos commented on subsys/net/lib/mqtt/mqtt_transport_socket_tls.c:
```diff
@@ -44,15 +44,15 @@ int mqtt_client_tls_connect(struct mqtt_client *client)
 	/* Set secure socket options. */
-	ret = setsockopt(client->transport.tls.sock, SOL_TLS, TLS_PEER_VERIFY,
+	ret = zsock_setsockopt(client->transport.tls.sock, SOL_TLS, TLS_PEER_VERIFY,
 			 &tls_config->peer_verify,
```
Here.

rlubos commented on subsys/net/lib/mqtt/mqtt_transport_socket_tls.c:
```diff
@@ -44,15 +44,15 @@ int mqtt_client_tls_connect(struct mqtt_client *client)
 	if (tls_config->cipher_list != NULL && tls_config->cipher_count > 0) {
-		ret = setsockopt(client->transport.tls.sock, SOL_TLS,
+		ret = zsock_setsockopt(client->transport.tls.sock, SOL_TLS,
 				 TLS_CIPHERSUITE_LIST, tls_config->cipher_list,
```
Here.

rlubos commented on subsys/net/lib/mqtt/mqtt_transport_socket_tls.c:
```diff
@@ -61,7 +61,7 @@ int mqtt_client_tls_connect(struct mqtt_client *client)
 	if (tls_config->sec_tag_list != NULL && tls_config->sec_tag_count > 0) {
-		ret = setsockopt(client->transport.tls.sock, SOL_TLS,
+		ret = zsock_setsockopt(client->transport.tls.sock, SOL_TLS,
 				 TLS_SEC_TAG_LIST, tls_config->sec_tag_list,
```
Here.

rlubos commented on subsys/net/lib/mqtt/mqtt_transport_socket_tls.c:
```diff
@@ -70,7 +70,7 @@ int mqtt_client_tls_connect(struct mqtt_client *client)
 	if (tls_config->hostname) {
-		ret = setsockopt(client->transport.tls.sock, SOL_TLS,
+		ret = zsock_setsockopt(client->transport.tls.sock, SOL_TLS,
 				 TLS_HOSTNAME, tls_config->hostname,
```
Here.

rlubos commented on subsys/net/lib/mqtt/mqtt_transport_socket_tls.c:
```diff
@@ -84,7 +84,7 @@ int mqtt_client_tls_connect(struct mqtt_client *client)
 
-	ret = connect(client->transport.tls.sock, client->broker,
+	ret = zsock_connect(client->transport.tls.sock, client->broker,
 		      peer_addr_size);
```
Here.

rlubos commented on subsys/net/lib/mqtt/mqtt_transport_socket_tls.c:
```diff
@@ -105,7 +105,7 @@ int mqtt_client_tls_write(struct mqtt_client *client, const uint8_t *data,
 	while (offset < datalen) {
-		ret = send(client->transport.tls.sock, data + offset,
+		ret = zsock_send(client->transport.tls.sock, data + offset,
 			   datalen - offset, 0);
```
And here.

xhpohanka replied:
oh, ok
fixed in new version

rlubos reviewed (commented):
Looks good, but there are several function calls that got their alignment broken after the rename, would be good to fix this.

rlubos reviewed (approved):
Thanks

MaureenHelm commented:
please review the buildkite failure

xhpohanka committed:
net: mqtt: fix sample to use zsock_ functions

Use zsock_ variants of socket functions in mqtt sample application.

Signed-off-by: Jan Pohanka <xhpohanka@gmail.com>

xhpohanka committed:
net: mqtt: fix tests to use zsock_ functions

Use zsock_ variants of socket functions in mqtt tests.

Signed-off-by: Jan Pohanka <xhpohanka@gmail.com>

xhpohanka committed:
net: sockets: tls: fix using of zsock_ functions

Use zsock_ variants of socket functions to be independent on
NET_SOCKETS_POSIX_NAMES config.

Signed-off-by: Jan Pohanka <xhpohanka@gmail.com>

xhpohanka committed:
drivers: wifi: fix simplelink driver to use zsock_ functions

This fix removes dependency on CONFIG_POSIX_API
or NET_SOCKETS_POSIX_NAMES.

Signed-off-by: Jan Pohanka <xhpohanka@gmail.com>

jukkar commented:
2.4 is released so we could try to merge this, @xhpohanka could you check the sanitychecker failures and re-submit?

xhpohanka commented:
> 2.4 is released so we could try to merge this, @xhpohanka could you check the sanitychecker failures and re-submit?

I was on holiday for last three weeks, I hope I will have time to fix it soon. From the log it seems to me that current sanity checks can be caused by something different than my patches, is that even possible?

jukkar commented:
>  is that even possible?

Yes, that is possible. Could you rebase against latest master and re-submit, we will then see if the errors will disappear.
